website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+
+## Custom Metrics: Tailoring Evaluation to Your Needs
+
+MLflow provides two main functions for creating custom metrics using LLMs as judges:
+
+1. [mlflow.metrics.genai.make_genai_metric()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.make_genai_metric)
+2. [mlflow.metrics.genai.make_genai_metric_from_prompt()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.make_genai_metric_from_prompt)
+
+Let's see how Worldwide WanderAgency can use these functions to create custom metrics for their translation evaluation.
+
+### Cultural Sensitivity Metric
+
+The travel agency wants to ensure their translations are not only accurate but also culturally appropriate. Here's how they could create a custom `"cultural_sensitivity"` metric
+
+This custom metric allows Worldwide WanderAgency to quantify how well their translations maintain cultural context and idiomatic expressions.
+
+### The Faithfulness Metric
+
+As Worldwide WanderAgency's AI capabilities grow, they decide to implement an AI-powered customer service chatbot to handle inquiries in multiple languages.
+
+After making sure their materials are well translated.
+The company decided to build a pipline for their materials which is invoked inside a `RAG` system.
+
+ To ensure the chatbot provides accurate information based on the company's policies, they create another custom metric: the `"faithfulness"` metric.
+
+This metric assesses how well the chatbot's responses align with the provided context (company policies) and the specific questions asked by customers.
+
+### Built in Metrics
+
+In addition to this we can leverage mlflow built in metrics to measure some relevant metrics.
+Since Worldwide WanderAgency uses a text model we get these metrics out of the box. 
+
+- [toxicity](https://huggingface.co/spaces/evaluate-measurement/toxicity) 
+- [ari_grade_level](https://en.wikipedia.org/wiki/Automated_readability_index) 
+- [flesch_kincaid_grade_level](https://en.wikipedia.org/wiki/Flesch%E2%80%93Kincaid_readability_tests#Flesch%E2%80%93Kincaid_grade_level) 
+
+For this use case we want to make sure none of the context translated are toxic.
+
+
+## Putting It All Together: Evaluating Worldwide WanderAgency's AI Systems
+
+With these custom metrics in place, the travel agency can now use MLflow Evaluate to assess both their translation model and customer service chatbot. They can track performance over time, identify areas for improvement, and ensure that their AI-driven communication maintains high standards of cultural sensitivity and accuracy.
+
+Here's a glimpse of how they might visualize their evaluation results:
+
+![Gauge Chart](gauge.png)
+
+This dashboard allows Worldwide WanderAgency's team to quickly assess the performance of their AI systems across different metrics, ensuring they're meeting the high standards required for effective global communication.
```

Comment:
This is largely repeated below. Suggest cutting this and jumping into the example here. "Now that we understand what WanderAgency is trying to measure and why it is challenging, let's walk through an example showing how to implement the types of measures we discussed above..."